### PR TITLE
Enable PRC by default in pulumi-azure

### PR DESCRIPTION
**Pull request by VenelinMartinov**
Enables PlanResourceChange by default in pulumi-azure.

Part of https://github.com/pulumi/pulumi-terraform-bridge/issues/1785

Also contains https://github.com/pulumi/pulumi-azure/pull/2325
Also contains https://github.com/pulumi/pulumi-azure/pull/2331

fixes https://github.com/pulumi/pulumi-azure/issues/2322
fixes https://github.com/pulumi/pulumi-azure/issues/2323
fixes https://github.com/pulumi/pulumi-azure/issues/2330

### CODE REVIEW

**Comment by t0yv0** [comment]
Really appreciate diagnosing test failures here. Are these semantic failures from the change or unrelated flakes ?

File: provider/resources.go [hunk 1]
```diff
@@ -502,9 +502,9 @@ var metadata []byte
 //
 // nolint: lll
 func Provider() tfbridge.ProviderInfo {
-	p := shimv2.NewProvider(shim.NewProvider(), shimv2.WithPlanResourceChange(func(tfResourceType string) bool {
-		return tfResourceType == "azurerm_storage_account"
-	}))
+	p := shimv2.NewProvider(shim.NewProvider(),
+		shimv2.WithPlanResourceChange(func(_ string) bool { return true }),
```
Comment (iwahbe) [inline provider/resources.go:506]: Nit: You can drop the `_` in the function signature.
```suggestion
		shimv2.WithPlanResourceChange(func(string) bool { return true }),
```

File: provider/resources.go [hunk 2]
```diff
@@ -1891,6 +1891,14 @@ func Provider() tfbridge.ProviderInfo {
...
 					}),
 				},
+				TransformFromState: func(_ context.Context, pm resource.PropertyMap) (resource.PropertyMap, error) {
+					// if the defaultOutboundAccessEnabled property is not set, set it to the default value of true
+					// this prevents an unnecessary replacement when upgrading the provider
+					if _, ok := pm["defaultOutboundAccessEnabled"]; !ok {
+						pm["defaultOutboundAccessEnabled"] = resource.NewBoolProperty(true)
+					}
+					return pm, nil
+				},
```
Comment (iwahbe) [inline provider/resources.go:1901]: Wasn't this included in https://github.com/pulumi/pulumi-azure/pull/2325? A rebase onto master might show the effects of this PR more clearly.

**Review by iwahbe — APPROVED** [review]
LGTM

File: provider/resources.go (hunk 2 above)
Comment (VenelinMartinov) [reply provider/resources.go:1901]: I've merged 2325 here, not in master - I didn't want to consider the effects of 2325 under non-PRC.

**Comment by VenelinMartinov** [comment]
There's two more test failures to be addressed before this can be merged, tracking in https://github.com/pulumi/pulumi-azure/issues/2322

**Review by t0yv0 — APPROVED** [review]

**Pull request merged**